JavaScript/JavaScriptSDK.Tests/CheckinTests/Sender.tests.ts

```diff
@@ -38,12 +38,19 @@ class SenderTests extends TestClass {
             emitLineDelimitedJson: () => this.emitLineDelimitedJson,
             maxBatchSizeInBytes: () => this.maxBatchSizeInBytes,
             maxBatchInterval: () => this.maxBatchInterval,
-            disableTelemetry: () => this.disableTelemetry
+            disableTelemetry: () => this.disableTelemetry,
+            storeSendBufferInSessionStorage: () => false
         };
 
-        this.getSender = () => new Microsoft.ApplicationInsights.Sender(config);
-        this.errorSpy = this.sandbox.spy(Microsoft.ApplicationInsights.Sender, "_onError");
-        this.successSpy = this.sandbox.spy(Microsoft.ApplicationInsights.Sender, "_onSuccess");
+        this.getSender = () => {
+            var sender = new Microsoft.ApplicationInsights.Sender(config);
+
+            this.errorSpy = this.sandbox.spy(sender, "_onError");
+            this.successSpy = this.sandbox.spy(sender, "_onSuccess");
```

Comment:
Just to make sure I understand, errorSpy and successSpy won't actually be defined until getSender is called?
It appears you want a new sender for each call to getSender in which case it makes sense to define errorSpy and successSpy relative to the new sender.
However, it is strange that errorSpy and successSpy are properties on the calling instance since each call to getSender will change the this. assignments and orphan previous instances.